os/net/local/local_connect.c

```diff
@@ -0,0 +1,346 @@
+ * ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ * POSSIBILITY OF SUCH DAMAGE.
+ *
+ ****************************************************************************/
+
+/****************************************************************************
+ * Included Files
+ ****************************************************************************/
+
+#include <tinyara/config.h>
+
+#include <string.h>
+#include <unistd.h>
+#include <assert.h>
+#include <errno.h>
+#include <queue.h>
+#include <debug.h>
+
+#include <tinyara/semaphore.h>
+#include <tinyara/net/net.h>
+
+#include <arch/irq.h>
+
+#include "utils/utils.h"
+#include "socket/socket.h"
+#include "local/local.h"
+
+#ifdef CONFIG_NET_LOCAL_STREAM
+
+/****************************************************************************
+ * Private Functions
+ ****************************************************************************/
+
+/****************************************************************************
+ * Name: local_generate_instance_id
+ ****************************************************************************/
+
+static int32_t local_generate_instance_id(void)
+{
+	static int32_t g_next_instance_id = 0;
+	int32_t id;
+
+	/* Called from local_connect with net_lock held. */
+
+	id = g_next_instance_id++;
+	if (g_next_instance_id < 0) {
+		g_next_instance_id = 0;
+	}
+
+	return id;
+}
+
+/****************************************************************************
+ * Name: _local_semtake() and _local_semgive()
+ *
+ * Description:
+ *   Take/give semaphore
+ *
+ ****************************************************************************/
+
+static inline void _local_semtake(sem_t *sem)
+{
+	net_lockedwait_uninterruptible(sem);
+}
+
+#define _local_semgive(sem) sem_post(sem)
+
+/****************************************************************************
+ * Name: local_stream_connect
+ *
+ * Description:
+ *   Find a local connection structure that is the appropriate "server"
+ *   connection to be used with the provided "client" connection.
+ *
+ * Returned Value:
+ *   Zero (OK) returned on success; A negated errno value is returned on a
+ *   failure.  Possible failures include:
+ *
+ * Assumptions:
+ *   The network is locked on entry, unlocked on return.  This logic is
+ *   an integral part of the lock_connect() implementation and was
+ *   separated out only to improve readability.
+ *
+ ****************************************************************************/
+
+static int inline local_stream_connect(FAR struct local_conn_s *client,
+									   FAR struct local_conn_s *server,
+									   bool nonblock)
+{
+	int ret;
+	int sval;
+
+	/* Has server backlog been reached?
+	 * NOTE: The backlog will be zero if listen() has never been called by the
+	 * server.
+	 */
+
+	if (server->lc_state != LOCAL_STATE_LISTENING ||
+		server->u.server.lc_pending >= server->u.server.lc_backlog) {
+		net_unlock();
```

Comment:
net_unlock(...) require one parameter (even though it's not used).
```c
void net_unlock(net_lock_t flags)
```

Reply:
I replaced `net_unlock()` with `net_unlock(0)`